internal/scorecard/alpha/options.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package alpha
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+
+	registryimage "github.com/operator-framework/operator-registry/pkg/image"
+	"github.com/operator-framework/operator-registry/pkg/image/containerdregistry"
+)
+
+// RunnerOptions helps construct a TestRunner by holding information that a
+// TestRunner may not need once created but will need during instantiation.
+type RunnerOptions struct {
+	Namespace      string
+	ServiceAccount string
+	KubeconfigPath string
+	Bundle         string
+}
+
+// NewPodTestRunner creates a new PodTestRunner from RunnerOptions.
+func (opts RunnerOptions) NewPodTestRunner(ctx context.Context, logger *log.Entry) (r *PodTestRunner, err error) {
```

Comment:
1, 2, 3: agreed.
4. Given when the config is currently [required](https://github.com/operator-framework/operator-sdk/pull/3009/files#diff-aeb4aa8564b377bc4a07e73cec100e36R85), bundle expansion needs to happen before `r.Initialize()`. This could probably be re-organized within this PR, although I'll need to check if parsing the config after bundle initialization makes sense. @jmccormick2001 is that reasonable?

Reply:
I think that is coded that way because both ListTests and RunTests both need the config to select tests from, so doing it in cmd.go effectively allowed the config to be passed down to both of those functions.

Reply:
Ah, and we're going to parse the config out of the bundle, so we need the bundle to load the config. Am I following that correctly?

If so, would it make sense to make the bundle extraction and loading a totally separate step that occurs first and then we hand the pod test runner a path to the bundle or a struct representing the bundle or whatever?

Reply:
Correct. Bundle extraction must always precede config loading. I think this can be rewritten as you suggest.

Reply:
Bundle extraction is now separate from runner construction. Also I've removed `PodRunnerOptions`, which we can implement separately.